src/banman.cpp

```diff
@@ -39,6 +39,10 @@ BanMan::~BanMan()
 
 void BanMan::DumpBanlist()
 {
+    static Mutex dump_mutex;
+    TRY_LOCK(dump_mutex, dump_lock);
+    if (!dump_lock) return;
```

Comment:
why is is ok to refuse to dump when two threads call this function?

Reply:
It is an optimization. Calling this function in the destructor guarantees that thr final state will be dumped to the disk.

Reply:
Then it should be mentioned in the release notes that calling the ban rpc no longer implies the stuff is written to disk and when the node crashes, the state will be lost.

Reply:
Not relevant since the recent [update](https://github.com/bitcoin/bitcoin/pull/24168#issuecomment-1024569483), I guess.